File: hs/right.hs
import System.Environment

right :: (Ord f, Enum f, Floating f) => f -> [(f, f, f)]
right hyp = [(a, b, c) |
   let r = [1..hyp],
   a <- r, b <- r,
   let c = sqrt (a^2 + b^ 2),
   c <= hyp]

intstr :: RealFrac a => a -> String
intstr = show . floor

printTriangle (a, b, c) = putStrLn $ intstr a ++ ", " ++ intstr b ++ ", " ++ show c

main = do
   args <- getArgs
   mapM_ printTriangle $ right $ read $ head args

File: py/right.py
import math

def right(hyp):
   r = range(1, hyp)
   return [
      (a, b, int(c))
      for a in r
      for b in r
      if (c := math.sqrt(a ^ 2 + b ^ 2)) <= hyp and c.is_integer()
   ]

def printTriangle(t):
   (a, b, c) = t;
   print("%s, %s, %s"%(a, b, c))

import sys

hypInput = int(sys.argv[1])
triangles = right(hypInput)

for t in triangles:
   printTriangle(t)
